src/modules/commander/Commander.cpp

```diff
@@ -3888,203 +3829,183 @@ bool Commander::preflight_check(bool report)
-			if (telemetry.type == telemetry_status_s::TELEMETRY_STATUS_RADIO_TYPE_USB) {
-				status_flags.usb_connected = true;
-			}
+						if (iridiumsbd_updated) {
+							iridiumsbd_status_s iridium_status;
 
-			/* set latency type of the telemetry connection */
-			if (telemetry.type == telemetry_status_s::TELEMETRY_STATUS_RADIO_TYPE_IRIDIUM) {
-				_telemetry[i].high_latency = true;
+							if (orb_copy(ORB_ID(iridiumsbd_status), _iridiumsbd_status_sub, &iridium_status) == PX4_OK) {
+								_high_latency_datalink_heartbeat = iridium_status.last_heartbeat;
 
-			} else {
-				_telemetry[i].high_latency = false;
-			}
+								if (status.high_latency_data_link_lost) {
+									if (hrt_elapsed_time(&_high_latency_datalink_lost) > (_high_latency_datalink_regain_threshold.get() * 1_s)) {
+										status.high_latency_data_link_lost = false;
+										status_changed = true;
+									}
+								}
 
-			if (telemetry.heartbeat_time > 0 && (_telemetry[i].last_heartbeat < telemetry.heartbeat_time)) {
-				_telemetry[i].last_heartbeat = telemetry.heartbeat_time;
-			}
-		}
-
-		// for iridium telemetry use the iridiumsbd_status to update the heartbeat
-		if (_telemetry[i].high_latency) {
-			if (_iridiumsbd_status_sub.update()) {
-				const hrt_abstime isbd_timestamp = _iridiumsbd_status_sub.get().last_heartbeat;
+							}
+						}
+					}
 
-				if (isbd_timestamp > _telemetry[i].last_heartbeat) {
-					_telemetry[i].last_heartbeat = isbd_timestamp;
+					break;
 				}
-			}
-		}
-	}
-}
 
-void Commander::data_link_checks(int32_t highlatencydatalink_loss_timeout, int32_t highlatencydatalink_regain_timeout,
-				 int32_t datalink_loss_timeout, int32_t datalink_regain_timeout, bool *status_changed)
-{
-	/* data links check */
-	bool have_link = false;
-	bool high_latency_link_exists = false;
-	bool have_low_latency_link = false;
-	int32_t dl_loss_timeout_local = 0;
-	int32_t dl_regain_timeout_local = 0;
 
-	for (int i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
-		if (_telemetry[i].high_latency) {
-			high_latency_link_exists = true;
+				// handle different remote types
+				switch (telemetry.remote_type) {
+				case (telemetry_status_s::MAV_TYPE_GCS):
+					_datalink_last_heartbeat_gcs = telemetry.heartbeat_time;
 
-			if (status.high_latency_data_link_active) {
-				dl_loss_timeout_local = highlatencydatalink_loss_timeout;
-				dl_regain_timeout_local = highlatencydatalink_regain_timeout;
-
-			} else {
-				// if the high latency link is inactive we do not want to accidentally detect it as an active link
-				dl_loss_timeout_local = INT32_MIN;
-				dl_regain_timeout_local = INT32_MAX;
-			}
+					// Recover from data link lost
+					if (status.data_link_lost) {
+						if (hrt_elapsed_time(&_datalink_last_heartbeat_gcs) < (_datalink_regain_threshold.get() * 1_s)) {
+							status.data_link_lost = false;
+							status_changed = true;
+							mavlink_log_critical(&mavlink_log_pub, "DATA LINK REGAINED");
```

Comment:
Please go through all messages and avoid capitalizing them.
```suggestion
							mavlink_log_critical(&mavlink_log_pub, "Data link regained");
```

Also a regain does not have to be critical IMO.

Reply:
For this suggestion and the ones below see here 4c37aa28d9453a0d05af08f8d8d239f98973adfd